src/main/java/fm/last/moji/impl/FileUploadOutputStream.java

```diff
@@ -80,40 +81,62 @@ public void flush() throws IOException {
   @Override
   public void close() throws IOException {
     log.debug("Close called on {}", this);
-    long size = -1L;
+
     try {
+      // Step 1. finish data writing to store node
```

Comment:
Step 1 could be factored into an internal private method, maybe something like flushAndClose() and then step 2 could go into another internal private method, maybe something like closeTracker()? Would make the code easier to read as this method is now quite long.